### PR TITLE
feature(amp-live-list): add update feature

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cramforce PTAL

**Comment by erwinmombay** [comment]
@cramforce ping

File: build-system/server.js [hunk 1]
```diff
@@ -103,6 +104,28 @@ app.use('/examples.build/live-list.amp.max.html', function(req, res) {
...
+var liveListUpdateFullPath = `${process.cwd()}${liveListUpdateFile}`;
+var liveListFile = fs.readFileSync(liveListUpdateFullPath);
+var ctr = 0;
+var doc = jsdom.jsdom(liveListFile);
+var win = doc.defaultView;
+app.use(liveListUpdateFile, function(req, res) {
+  var doctype = '<!doctype html>\n';
+  res.setHeader('Content-Type', 'text/html');
+  res.statusCode = 200;
+  if (ctr != 0) {
```
Comment (cramforce) [inline build-system/server.js]: Should this be throttled a bit and sometimes return nothing?

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ export class LiveListInterface {
    * Update the underlying live list dom structure.
    *
    * @param {?Element} element
+   * @return {time}
```
Comment (cramforce) [inline extensions/amp-live-list/0.1/amp-live-list.js]: We typically use number or is there a typedef?

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 3]
```diff
@@ -195,33 +198,125 @@ export class AmpLiveList extends AMP.BaseElement {
...
+          this.viewport_./*OK*/scrollIntoView(this.element);
+        });
       });
+    }
+    return promise;
+  }
+
+  /**
+   * @param {!Element} parent
+   * @param {!Array<!Element>} items
```
Comment (cramforce) [inline extensions/amp-live-list/0.1/amp-live-list.js]: orphans?

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 4]
```diff
@@ -195,33 +198,125 @@ export class AmpLiveList extends AMP.BaseElement {
...
+        this.getVsync().mutate(() => {
+          // Should scroll into view be toggleable
+          this.viewport_./*OK*/scrollIntoView(this.element);
+        });
       });
+    }
+    return promise;
+  }
+
+  /**
```
Comment (cramforce) [inline extensions/amp-live-list/0.1/amp-live-list.js]: Please add a few docs here and elsewhere.

**Comment by cramforce** [comment]
LGTM

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 2 above)
Comment (erwinmombay) [inline extensions/amp-live-list/0.1/amp-live-list.js]: I don't think so. I think I originally used number and reviewed a PR from @dvoytenko using time and switched over.

Comment (dvoytenko) [inline extensions/amp-live-list/0.1/amp-live-list.js]: We do have a typedef and I usually ask everyone to use it. Although, if we decide it's not useful, let's remove.

Comment (erwinmombay) [inline extensions/amp-live-list/0.1/amp-live-list.js]: @dvoytenko thanks for correcting. IMO, i think its useful. @cramforce i think you mentioned this sort of aliasing doesn't work with CC right? (correct me if I'm wrong)

Comment (cramforce) [inline extensions/amp-live-list/0.1/amp-live-list.js]: Its great. `time` is much better than `number`.

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 4 above)
Comment (erwinmombay) [inline extensions/amp-live-list/0.1/amp-live-list.js]: done. added on all new functions

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 3 above)
Comment (erwinmombay) [inline extensions/amp-live-list/0.1/amp-live-list.js]: done.

File: build-system/server.js (hunk 1 above)
Comment (erwinmombay) [inline build-system/server.js]: good point, will add variation here and on upcoming full sample

Comment (erwinmombay) [inline build-system/server.js]: done.

**Pull request merged**